homeassistant/components/shelly/__init__.py

```diff
@@ -216,6 +218,22 @@ def _async_device_online(_: Any, update_type: BlockUpdateType) -> None:
             raise ConfigEntryNotReady(repr(err)) from err
         except InvalidAuthError as err:
             raise ConfigEntryAuthFailed(repr(err)) from err
+        except FirmwareUnsupported as err:
+            error = repr(err)
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                FIRMWARE_UNSUPPORTED_ISSUE_ID.format(unique=entry.unique_id),
+                is_fixable=False,
```

Comment:
We try not to create non fixable issues and usually this is only a last resort when there is no solution to fix it for them.

In this case I think we could check if an update is available and the fix path could be to apply the update.  

It might be easier to do a post to [`/rpc`](https://shelly-api-docs.shelly.cloud/gen2/General/RPCChannels#post-request-with-entire-frame-as-payload) for gen2 so we don't have to setup a full websocket connection and than the only thing we ever have to keep backwards compatible between major firmwares is the ability to update

Reply:
> In this case I think we could check if an update is available and the fix path could be to apply the update.

The first part is basically the same as #108122

Reply:
and is valid for all generation devices